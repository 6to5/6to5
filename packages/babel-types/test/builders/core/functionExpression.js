import * as t from "../../..";

describe("builders", function () {
  describe("core", function () {
    describe("functionExpression", function () {
      it("accept all parameters", function () {
        const functionExpression = t.functionExpression(
          t.identifier("foo"),
          [t.identifier("bar")],
          t.blockStatement([t.returnStatement(t.thisExpression())]),
          true,
          true,
          t.tsTypeAnnotation(t.tsStringKeyword()),
          t.tsTypeParameterDeclaration([
            t.tsTypeParameter(t.tsObjectKeyword(), null, "bar"),
          ]),
        );
        expect(functionExpression).toMatchSnapshot();
      });
    });
  });
});
