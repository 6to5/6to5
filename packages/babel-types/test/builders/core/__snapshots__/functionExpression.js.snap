// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builders core functionExpression accept all parameters 1`] = `
Object {
  "async": true,
  "body": Object {
    "body": Array [
      Object {
        "argument": Object {
          "type": "ThisExpression",
        },
        "type": "ReturnStatement",
      },
    ],
    "directives": Array [],
    "type": "BlockStatement",
  },
  "generator": true,
  "id": Object {
    "name": "foo",
    "type": "Identifier",
  },
  "params": Array [
    Object {
      "name": "bar",
      "type": "Identifier",
    },
  ],
  "returnType": Object {
    "type": "TSTypeAnnotation",
    "typeAnnotation": Object {
      "type": "TSStringKeyword",
    },
  },
  "type": "FunctionExpression",
  "typeParameters": Object {
    "params": Array [
      Object {
        "constraint": Object {
          "type": "TSObjectKeyword",
        },
        "default": null,
        "name": "bar",
        "type": "TSTypeParameter",
      },
    ],
    "type": "TSTypeParameterDeclaration",
  },
}
`;
